File: 백준/Silver/11723. 집합/집합.py
import sys
input = sys.stdin.readline

M = int(input())
S = set()
S = [0] * 20

for _ in range(M) :
    cmds = list(input().split())
    command = cmds[0]
    if len(cmds) == 1 :
        if command == 'all' :
            for i in range(20) :
                S[i] = 1
        elif command == 'empty' :
            for i in range(20) :
                S[i] = 0
    elif len(cmds) == 2 :
        param = int(cmds[1])
        if command == 'add' :
            S[param - 1] = 1
        elif command == 'remove' :
            S[param - 1] = 0
        elif command == 'check' :
            if S[param - 1] : print(1)
            else : print(0)
        elif command == 'toggle' :
            if S[param - 1] : S[param - 1] = 0
            else : S[param - 1] = 1

File: 백준/Silver/11723. 집합/집합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int M = Integer.parseInt(br.readLine());
        int x = 0;
        int s = 0;
        for (int i = 0; i < M; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String cmd = st.nextToken();
            if (st.hasMoreTokens()) x = Integer.parseInt(st.nextToken());
            if (cmd.equals("add")){
                s |= 1 << (x - 1);
            } else if (cmd.equals("remove")) {
                s &= ~(1 << (x - 1));
            } else if (cmd.equals("check")){
                sb.append(( (s & (1 << (x - 1)) ) >> (x - 1)) + "\n");
            } else if (cmd.equals("toggle")){
                if (( (s & (1 << (x - 1)) ) >> (x - 1)) == 1){
                    s &= ~(1 << (x - 1));
                } else{
                    s |= 1 << (x - 1);
                }
            } else if (cmd.equals("all")){
                s = -1;
            } else if (cmd.equals("empty")) {
                s = 0;
            }
        }
        System.out.println(sb);
    }
}